package NYT.NFileClient.NProto;

import "yt/ytlib/misc/guid.proto";
import "yt/ytlib/cypress_client/cypress_ypath.proto";
import "yt/ytlib/node_tracker_client/node.proto";
import "yt/ytlib/chunk_client/chunk.proto";

////////////////////////////////////////////////////////////////////////////////

// Create extension.

message TReqCreateFileExt
{
    extend NCypressClient.NProto.TReqCreate
    {
        optional TReqCreateFileExt create_file_ext = 100;
    }

    required NYT.NProto.TGuid chunk_id = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqPrepareForUpdate
{ }

message TRspPrepareForUpdate
{
    required NYT.NProto.TGuid chunk_list_id = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqFetchFile
{ }

message TRspFetchFile
{
    required NYT.NProto.TGuid chunk_id = 1;
    required NYT.NNodeTrackerClient.NProto.TNodeDirectory node_directory = 2;
    repeated fixed32 replicas = 3;
    required string file_name = 4;
    required bool executable = 5;
}

////////////////////////////////////////////////////////////////////////////////
